docs/operational_guide/bootstrapping.md

```diff
@@ -0,0 +1,112 @@
+
+### Peers Bootstrapper
+
+The peer bootstrapper's responsibility is to stream in data for shard/ranges from other M3DB nodes (peers) in the cluster. This bootstrapper is only useful in M3DB clusters with more than a single node *and* where the replication factor is set to a value larger than 1. The `peers` bootstrapper will determine whether or not it can satisfy a bootstrap request on a shard-by-shard basis by consulting the cluster placement and determining if there are enough peers to satisfy the bootstrap request. For example, imagine the following M3DB placement where node 1 is trying to perform a peer bootstrap:
+
+```
+    ┌─────────────────┐          ┌─────────────────┐        ┌─────────────────┐
+    │     Host 1      │          │     Host 2      │        │     Host 3      │
+────┴─────────────────┴──────────┴─────────────────┴────────┴─────────────────┴───
+┌─────────────────────────┐   ┌───────────────────────┐   ┌──────────────────────┐
+│                         │   │                       │   │                      │
+│                         │   │                       │   │                      │
+│  Shard 1: Initializing  │   │ Shard 1: Initializing │   │  Shard 1: Available  │
+│  Shard 2: Initializing  │   │ Shard 2: Initializing │   │  Shard 2: Available  │
+│  Shard 3: Initializing  │   │ Shard 3: Initializing │   │  Shard 3: Available  │
+│                         │   │                       │   │                      │
+│                         │   │                       │   │                      │
+└─────────────────────────┘   └───────────────────────┘   └──────────────────────┘
+```
+
+In this case, the peer bootstrapper running on node 1 will not be able to fullfill any requests because node 2 is in the `Initializing` state for all of its shards and cannot fulfill bootstrap requests. This means that node 1's peer bootstrapper cannot meet its default consistency level of majority for bootstrapping (1 < 2 which is majority with a replication factor of 3). On the other hand, node 1 would be able to peer bootstrap in the following placement because its peers (nodes 2/3) are `Available` for all of their shards:
```

Comment:
one more subtle point - the peers in peers bootstrapping refers to bootstrapping from shard replicas owned by peers. Need to update the last sentence to clarify this

**old**
On the other hand, node 1 would be able to peer bootstrap in the following placement because its peers (nodes 2/3) are `Available` for all of their shards:

**new**
On the other hand, node 1 would be able to peer bootstrap it's shards in the following placement because its peers (nodes 2/3) have sufficient replicas of the shards it needs in the `Available` state:

Reply:
much cleaner, thanks